spec/services/step_nav_publisher_spec.rb

```diff
@@ -24,57 +24,23 @@
     context "published step by step having a step edited" do
       let(:step_nav) { create(:published_step_by_step_page) }
 
-      it "defines :auth_bypass_ids in the publishing-api payload" do
+      it "defines :access_limited in the publishing-api payload" do
+        step_nav.mark_draft_updated
         StepNavPublisher.update(step_nav)
-        content_id = /[0-9a-f]{5,40}/ # changes on each test run, as does Array value below
         payload = hash_including(:auth_bypass_ids => instance_of(Array))
-        expect(Services.publishing_api).to have_received(:put_content).with(content_id, payload)
+        expect(Services.publishing_api).to have_received(:put_content).with(step_nav.content_id, payload)
       end
     end
 
     context "step by step being edited" do
-      let(:step_nav) { create(:step_by_step_page_with_steps) }
-
-      before do
-        allow(SecureRandom).to receive(:uuid).and_return("bd814775-304c-46d0-919f-f9e0f6cba4e9")
-      end
-
-      it "does not revert to draft if a step by step in submitted_for_2i state is edited" do
-        step_nav.update_attributes(status: "submitted_for_2i", review_requester_id: review_requester_user.uid, reviewer_id: reviewer_user.uid)
-        StepNavPublisher.update(step_nav)
-        expect(step_nav.status).to eq "submitted_for_2i"
-      end
-
-      it "does not revert to draft if a step by step in in_review state is edited" do
-        step_nav.update_attributes(status: "in_review", review_requester_id: review_requester_user.uid, reviewer_id: reviewer_user.uid)
-        StepNavPublisher.update(step_nav)
-        expect(step_nav.status).to eq "in_review"
-      end
-
-      it "does not revert to draft if a step by step in approved_2i state is edited" do
-        step_nav.update_attributes(status: "approved_2i", review_requester_id: review_requester_user.uid, reviewer_id: reviewer_user.uid)
-        StepNavPublisher.update(step_nav)
-        expect(step_nav.status).to eq "approved_2i"
-      end
-
-      it "reverts to draft if a step by step in published state is edited" do
-        step_nav.update_attributes(status: "published")
-        StepNavPublisher.update(step_nav)
-        expect(step_nav.status).to eq "draft"
-      end
-
-      it "reverts to draft if a step by step in scheduled state is edited" do
-        step_nav.update_attributes(status: "scheduled")
-        StepNavPublisher.update(step_nav)
-        expect(step_nav.status).to eq "draft"
-      end
-
       it "does not change the auth_bypass_id when step by step in draft state is edited" do
         StepNavPublisher.update(step_nav)
         expect(step_nav.auth_bypass_id).to eq "33ac75d8-4adf-48a7-acb2-bbf4e7f644a3"
       end
 
       it "changes the auth_bypass_id when step by step in published state is edited" do
+        allow(SecureRandom).to receive(:uuid).and_return("bd814775-304c-46d0-919f-f9e0f6cba4e9")
```

Comment:
Could you move this content ID to its own variable, so we don't have to repeat the string on lines 42 and 46?

Reply:
Sure!